pym/bob/input.py

```diff
@@ -911,8 +911,91 @@ def hasJenkinsPlugin(self):
         return False
 
     def status(self, workspacePath, dir, verbose = 0):
-        print("CVS SCM status not implemented!")
-        return 'error'
+        workDir = os.path.join(workspacePath, dir)
+        cvsDir = os.path.join(workDir, 'CVS')
+        if not os.path.exists(workDir) or not os.path.exists(cvsDir):
+            return 'error'
+
+        # Prepare an environment
+        environment = os.environ.copy()
+        m = re.match('^:ssh:(.*)', self.__cvsroot)
+        if m:
+            expectedRoot = ":ext:" + m.group(1)
+            environment['CVS_RSH'] = 'ssh'
+        else:
+            expectedRoot = self.__cvsroot
+
+        # Validate root and module
+        status=""
+        longStatus=""
+
+        actualRoot = CvsScm._loadFile(os.path.join(cvsDir, 'Root'))
+        actualModule = CvsScm._loadFile(os.path.join(cvsDir, 'Repository'))
+
+        if actualRoot != expectedRoot:
+            # Root mismatch
+            status += "S"
+            longStatus += colorize("     > Root does not match!\n     recipe:\t{}\n     actual:\t{}\n".format(self.__cvsroot, expectedRoot), "33")
+        elif actualModule != self.__module:
+            # Module mismatch
+            status += "S"
+            longStatus += colorize("     > Module does not match!\n     recipe:\t{}\n     actual:\t{}\n".format(self.__module, actualModule), "33")
+        else:
+            # Repository matches.
+            # There is no (easy) local-only way to determine just local changes AND match it against the requested revision.
+            # We therefore just call 'cvs update' in "don't modify" mode and look what it would change.
+            # This will contain switched files, local and remote modifications.
+            try:
+                commandLine = ['cvs', '-Rnq', 'update', '-dP']
+                if self.__rev is None:
+                    commandLine += ['-A']
+                else:
+                    commandLine += ['-r', self.__rev]
+                output = subprocess.check_output(commandLine,
+                                                 cwd=workDir,
+                                                 universal_newlines=True,
+                                                 env=environment,
+                                                 stderr=subprocess.DEVNULL,
+                                                 stdin=subprocess.DEVNULL)
+            except subprocess.CalledProcessError as e:
+                raise BuildError("cvs error: '{}' '{}'".format(" ".join(cmdLine), e.output))
```

Comment:
Just print the error and return `'error'`. The status command should not throw exceptions. Otherwise `--clean-checkout` wouldn't work and the user has the inconvenience of having to do something about it to get the status command to move on...